app/assets/javascripts/shipit/deploy.js.coffee

```diff
@@ -1,7 +1,32 @@
-$(document).on 'click', '.deploy-abort .deploy-action', (event) ->
-  event.preventDefault()
-  $target = $(event.currentTarget)
-  return if $target.hasClass('pending')
-  $target.addClass('pending')
-  enable = -> $target.removeClass('pending')
-  $.post($target.attr('href')).done(-> setTimeout(enable, 3000))
+class AbortButton
+  SELECTOR = '[data-action="abort"]'
+
+  @handle: (event) =>
+    event.preventDefault()
+    $element = $(event.currentTarget).closest(SELECTOR)
+    button = new this($element)
+    button.trigger()
+
+  constructor: (@$button) ->
+    @url = @$button.find('a[href]').attr('href')
+    @shouldRollback = @$button.data('rollback')
+
+  trigger: ->
+    return if @isDisabled()
+
+    @disable()
+    $.post(@url).success(@waitForCompletion).error(@reenable)
+
+  waitForCompletion: =>
+    setTimeout(@reenable, 3000)
+
+  reenable: =>
+    @$button.removeClass('pending')
+
+  disable: ->
+    @$button.addClass('pending')
+
+  isDisabled: ->
+    @$button.hasClass('pending')
+
+$(document).on('click', '[data-action=abort]', AbortButton.handle)
```

Comment:
Seems odd to me to build an `AbortButton` by passing `AbortButton.handle` as a callback. IMHO having something like `AbortButton.listen()` handle the binding would be cleaner.

Also, shouldn't this be `$(document).on('click', AbortButton.SELECTOR, AbortButton.handle)`?